pkg/k8s/resource/example/main.go

```diff
@@ -54,16 +53,20 @@ func main() {
 var resourcesCell = cell.Module(
 	"resources",
 	cell.Provide(
-		resource.NewResourceConstructor[*corev1.Pod](
-			func(c client.Clientset) cache.ListerWatcher {
-				return utils.ListerWatcherFromTyped[*corev1.PodList](c.CoreV1().Pods(""))
-			},
-		),
-		resource.NewResourceConstructor[*corev1.Service](
-			func(c client.Clientset) cache.ListerWatcher {
-				return utils.ListerWatcherFromTyped[*corev1.ServiceList](c.CoreV1().Services(""))
-			},
-		),
+		func(lc hive.Lifecycle, c client.Clientset) resource.Resource[*corev1.Pod] {
+			if !c.IsEnabled() {
+				return nil
+			}
+			lw := utils.ListerWatcherFromTyped[*corev1.PodList](c.CoreV1().Pods(""))
+			return resource.New[*corev1.Pod](lc, lw)
+		},
+		func(lc hive.Lifecycle, c client.Clientset) resource.Resource[*corev1.Service] {
+			if !c.IsEnabled() {
+				return nil
+			}
```

Comment:
If the client is not enabled then all the embedded clientsets in `Clientset` are nil (e.g. clientset.CoreV1() is a nil deref), so we need to check for that. I guess alternatively we could return Clientset as `nil` instead of having a "disabled" clientset. I think I was just replicating the `k8s.IsEnabled` without thinking too much.

And yes, I'm choosing here to have a "static" object graph that is not conditional on whether or not features are enabled by config. I think that's the safer approach as then we'll need to test only once that all dependencies are satisfied, and we can produce one dot graph visualization that includes everything instead of gazillion combinations.

We could build this in a way to only `Provide` when `Config.Enable` is true, but this would then require all dependees to use `cell.In` struct with `optional:"true"` struct tags and check if the field is nil.